### PR TITLE
Implemented ldap_search action in collections controller

### PR DESCRIPTION
**Pull request by tonmoy18**
Added div load ajax, modified ldap_search action and view

Added ajax call in js and action in the controller

Added the ajax inputs to creator partial

Added the ldap search to contributor field

Made text box text presist

Adding the first displayname only

### CODE REVIEW

File: app/controllers/collections_controller.rb [hunk 1]
```diff
@@ -102,5 +102,17 @@ def crossref_search
...
+  def ldap_search
+    ldap = Net::LDAP.new(host: 'directory.vt.edu')
+    ldap.bind
+    treebase = 'ou=People,dc=vt,dc=edu'
+    ldap_attributes = {'uid': :authid, 'display_name': :displayname, 'department': :department, 'address': :postaladdress}
+
+		name = params[:name].gsub(/\s/,'*')
+		filter = Net::LDAP::Filter.eq("cn", "*#{name}*")
+		@results = ldap.search(base: treebase, filter: filter)
+		render :layout => false
```
Comment (rotated8) [inline app/controllers/collections_controller.rb]: Please fix the whitespace here.

File: app/assets/javascripts/datarepo/submission_form.js [hunk 2]
```diff
@@ -68,4 +68,33 @@ Blacklight.onLoad(function () {
   });
 
 
+	$('#ldap_button_creator').on('click', function () {
+		$( ".ldap_results_creator" ).load( 'ldap_search?name=' + $( '#ldap_text_creator' ).val() );
+		$('.ldap_results_creator').show();
+		$('#add_ldap_creator').show();
+	});
```
Comment (rotated8) [inline app/assets/javascripts/datarepo/submission_form.js]: Whitespace in this file is inconsistent, please fix.

File: config/routes.rb [hunk 3]
```diff
@@ -2,6 +2,8 @@
   get 'collections/datacite_search', to: 'collections#datacite_search'
   post 'collections/crossref_search', to: 'collections#crossref_search'
   get "collections/import_metadata", to: 'collections#import_metadata', as: 'import_collection'
+  get 'collections/ldap_search', to: 'collections#ldap_search'
+
```
Comment (rotated8) [inline config/routes.rb]: Extra blank line. Please remove.

File: app/views/collections/ldap_search.html.erb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+          <td><%= result[:department].join(" or ") %> </td>
+          <td><%= result[:mail].join(" or ") %> </td>
+        </tr>
+      <% end %>
+      </tbody>
+    </table>
+  </div>
+<% end %>
+
+
```
Comment (rotated8) [inline app/views/collections/ldap_search.html.erb]: Extra blank line, please remove.

**Comment by rotated8** [comment]
Just some whitespace issues, then :+1:

**Comment by yinlinchen** [comment]
Which JIRA ticket does this PR belong to?

**Comment by rotated8** [comment]
138

**Comment by tonmoy18** [comment]
Fixed the white space issues

**Comment by rotated8** [comment]
:+1:

**Pull request merged**